### PR TITLE
Implements first iteration of multi diff editors.

### LINKED ISSUE #179000: Multi File Diff Editor
Explore creating a custom editor that would show a list of diff editor widget.

### PR DESCRIPTION
**Pull request by hediet**
Code_-_OSS_sWZYa7jmeT

Tasks left:
* More performant scrolling
* Collapsing files
* adding/removing files

Fixes #179000

### CODE REVIEW

File: src/vs/workbench/browser/parts/editor/editorCommands.ts [hunk 1]
```diff
@@ -632,6 +632,39 @@ function registerOpenEditorAPICommands(): void {
...
+			args: [
+				{ name: 'title', description: 'Human readable title for the diff editor' },
+				{ name: 'resources', description: 'List of resources to open in the changes editor' }
+			]
+		}
+	});
+
+	CommandsRegistry.registerCommand('_workbench.changes', async (accessor: ServicesAccessor, title: string, resources: [UriComponents, UriComponents?, UriComponents?][]) => {
+		const editorService = accessor.get(IEditorService);
+		// const editorGroupService = accessor.get(IEditorGroupsService);
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorCommands.ts:654]: 2x Commented out lines of code

File: src/vs/workbench/common/editor.ts [hunk 2]
```diff
@@ -489,6 +489,17 @@ export interface IResourceDiffEditorInput extends IBaseUntypedEditorInput {
...
 
+/**
+ * A resource list diff editor input compares multiple resources side by side
+ * highlighting the differences.
+ */
+export interface IResourceMultiDiffEditorInput extends IBaseUntypedEditorInput {
+	/**
+	 * The list of resources to compare.
+	 */
+	readonly resources: (IResourceDiffEditorInput & { resource: URI })[];
```
Comment (bpasero) [inline src/vs/workbench/common/editor.ts:500]: Should `resource` be `readonly resource: URI`?

**Review by bpasero — DISMISSED** [review]
I was only looking at a few files, not the larger pieces and left some feedback. I am adding @lramos15  for reviewing the editor resolver changes.

**Comment by hediet** [comment]
> I'm guessing there is, but just for my understanding is there a reasoning for not making this a one off editor type like Chat, Search, the Perf graph, etc instead of adding an entire new editor type that my guess is won't be implemented elsehwere

I was wondering the same. Let's start with this and refactor later.
Because the widget lives separately, it should be easy to address this later.

**Review by lramos15 — APPROVED** [review]

**Comment by IllusionMH** [comment]
Is it only for Git (VCS) view for now?
I can't find command that would allow me to use it to diff 2 folders. Are there plans to support folders diff?
Similar to how keybinding below allows to quickly diff files(or open editors)
```json
    {
        "key": "ctrl+d",
        "command": "compareSelected",
        "when": "listDoubleSelection && activeViewlet == 'workbench.view.explorer'"
    },
```

**Comment by hediet** [comment]
> Is it only for Git (VCS) view for now (at least this is my impression after glancing over this PR code)?

We will slowly integrate it into other features as well.

> I can't find command that would allow me to use it to diff 2 folders. Are there plans to support folders diff?

This is out of scope for this feature, but maybe this can be implemented by an extension.

**Pull request merged**